### DISCUSSION TITLE
Where can I find historical document?

### DISCUSSION THREAD

**Discussion by guanpu** [JedWatson/react-select#4917] [Q&A]
Sorry to raise such a discussion when it seems to be a simple question that should be solved in 10 seconds' search.
But I really spend sometime searching the website and issue list here, didn't find anything relevant. Yes, I find upgrade notes but that's not directly helpful as a historical document.
So how do I find documents for previous versions? To be specific I need V4.

**Comment by ebonow** [comment]
@guanpu Version 5 was a major version change in that the underlying Type library was changed from Flow to TS. Most everything else has remained unchanged from version 4 to version 5 and hasn't occurred as requiring its own api documentation , and the few "breaking changes" have been highlighted [here](https://github.com/JedWatson/react-select/releases/tag/react-select%405.0.0) but most of these changes are internal methods that wouldn't impact the external api or documentation.

**Reply by guanpu** [reply]
Thanks for you reply @ebonow 
The "DropdownIndicatorProps" to be specific, that I know from V5 documentation, doesn't exist in V4's types/react-select, nor mentioned in the upgrade doc as you linked here. So I suppose this is attributed as 'internal methods' as you said?
Unfortunately we have to do something complex that we have to use internals. If any V4 document exists, at least I can infer what's its equivalence of "DropdownIndicatorProps". 
And what's more important is I don't know how many other props is in such situation. It makes me not confident of adding feature in current implementation without upgrading to V5.

**Reply by ebonow** [reply]
I pulled up a copy of v4.2.1 to verify any differences. It seems that `isDisabled` was added to the `DropdownIndicatorProps`. There's been some discussion regarding better standardization of the "commonProps" especially with `innerProps`, `isDisabled` and `isFocused`.


innerProps is missing on the Input component as it's become standard that all extra props on the Input are automatically added as innerProps to the input element (hence the extremely long list of props being generated for this component).

isDisabled and isFocused are missing on some components because of a naming oversight of these props being relative to the Select or to the component (ie: Option/MultiValue) so it was decided to wait to have this added. 

However, it would seem though that the `isDisabled` was added at some point between 4.2.1 and the current version for the DropdownIndicatorProps so that seems to be the only difference I was able to find but here is a screenshot of the two Types definitions of 4.2.1 vs 5.2.1 in case I'm missing something else. 
![Screen Shot 2021-11-17 at 2 25 56 PM](https://user-images.githubusercontent.com/1007579/142294508-ed7fcf6a-af8c-400d-8ed0-38a63a2a144a.png)
![Screen Shot 2021-11-17 at 2 37 45 PM](https://user-images.githubusercontent.com/1007579/142294517-26f6766f-db36-49dd-a497-c136fe3d33e5.png)

Also note that the preview of the CommonProps in the v4 screenshot cut off "theme" so I added the text in this case.

Hope this helps, but please let us know if you have further questions that we could help with.

**Reply by guanpu** [reply]
Hi @ebonow It's not about whether 'isDisabled' is in the interface or not, it's about the interface name being "DropdownIndicatorProps" vs "IndicatorProps". So I see sample code using "DropdownIndicatorProps" from the document(which is V5), and I used it in my V4 code base, but clearly typescript won't make it compile, I have to dig into V4 source code to know I have to rename "DropdownIndicatorProps" into "IndicatorProps" to make it compile. What I am saying is this renaming change should be documented somewhere.
